frontend/public/components/limit-range.tsx

```diff
@@ -0,0 +1,173 @@
+  />;
+
+export const LimitRangeListPage: React.SFC<LimitRangeListPageProps> = props =>
+  <ListPage
+    {...props}
+    title="Limit Ranges"
+    kind={LimitRangeReference}
+    ListComponent={LimitRangeList}
+    canCreate={true}
+  />;
+
+const LimitRangeDetailsHeader = props =>
+  <ListHeader>
+    <ColHead {...props} className="col-xs-2" sortField="resourceType">Resource Type</ColHead>
+    <ColHead {...props} className="col-xs-2" sortField="min">Min</ColHead>
+    <ColHead {...props} className="col-xs-2" sortField="max">Max</ColHead>
+    <ColHead {...props} className="col-xs-2" sortField="defaultRequest">Default Request</ColHead>
+    <ColHead {...props} className="col-xs-2" sortField="default">Default Limit</ColHead>
+    <ColHead {...props} className="col-xs-2" sortField="maxLimitRequestRatio">Max Limit/Request Ratio</ColHead>
+  </ListHeader>;
+
+const LimitRangeDetailsRow: React.SFC<LimitRangeRowProps> = ({obj}) =>
+  <div className="row co-resource-list__item">
+    <div className="col-xs-2">
+      {obj.resourceType}
+    </div>
+    <div className="col-xs-2">
+      {obj.min || '-'}
+    </div>
+    <div className="col-xs-2">
+      {obj.max || '-'}
+    </div>
+    <div className="col-xs-2">
+      {obj.defaultRequest || '-'}
+    </div>
+    <div className="col-xs-2">
+      {obj.default || '-'}
+    </div>
+    <div className="col-xs-2">
+      {obj.maxLimitRequestRatio || '-'}
+    </div>
+  </div>;
+
+export const LimitRangeDetailsList = (resource) => {
+  const data = [];
+
+  for (let limitRange of resource.resource.spec.limits) {
+    let type = limitRange.type; //We have nested types, top level type is something like "Container"
+    let limitRangeProperties = ["max", "min", "default", "defaultRequest", "maxLimitRequestRatio"];
+    let cpuLimits = {};
+    let memoryLimits = {};
+    for (let index in limitRangeProperties) {
+      let limitRangeProperty = limitRangeProperties[index];
+      if (limitRange.hasOwnProperty(limitRangeProperty)){
+        if (limitRange[limitRangeProperty].hasOwnProperty("cpu")){
+          let value = limitRange[limitRangeProperty]["cpu"];
+          if (limitRangeProperty !== "maxLimitRequestRatio"){
+            // make sure cpu values have the correct units
+            // m = millicores, '' = cores
+            if (value.slice(-1) === 'm') {
+              value = value.slice(0, -1) + ' millicores';
+            } else {
+              value += " cores";
+            }
+          }
+          cpuLimits[limitRangeProperty] = value;
+        }
+        if (limitRange[limitRangeProperty].hasOwnProperty("memory")){
+          let value = limitRange[limitRangeProperty]["memory"]
+          if (limitRangeProperty !== "maxLimitRequestRatio") {
+            // make sure memory values have the correct units
+            // Mi = MiB, Gi = GiB, M = MB, G = GB
+            value += "B";
+          }
+          memoryLimits[limitRangeProperty] = value;
+        }
+      }
+    }
```

Comment:
If you don't add any special handling of the units, can we remove all this logic?

If we keep them, we should consider using lodash utilities like _.each, _.map, etc

Reply:
:+1: Can't believe `for` loops are obsolete these days :-(